mesh/ncmesh.hpp

```diff
@@ -428,17 +428,20 @@ class NCMesh
    /// Return the number of root elements.
    int GetNumRootElements() { return root_state.Size(); }
 
-   /// Return the distance of leaf 'i' from the root.
+   /// Return the distance of leaf @a i from the root.
    int GetElementDepth(int i) const;
 
    /** Return the size reduction compared to the root element (ignoring local
        stretching and curvature). */
    int GetElementSizeReduction(int i) const;
 
-   /// Return the faces and face attributes of leaf element 'i'.
+   /// Return the faces and face attributes of leaf element @a i.
    void GetElementFacesAttributes(int i, Array<int> &faces,
                                   Array<int> &fattr) const;
 
+   /// Set the attribute of leaf element @a i, which is a Mesh element index.
+   void SetAttribute(int i, int attr)
+   { elements[leaf_elements[i]].attribute = attr; }
```

Comment:
I think this is a reasonable change, but I think there was some issue about how to treat a mismatch in attribute between parent and child. Essentially if a parent with attribute 1, is refined, then the children all have different attributes, how to define what happens after coarsening?

I don't think this is a barrier though, as it should just be a bit buyer beware if a user is doing this manual attribution, or the user should not allow coarsening beyond this ambiguous point.

Reply:
One possibility is to make this a `protected` function. It is called by `Mesh::SetAttribute`, which is sufficient to fix at least one issue. However, `NCMesh` is usually not manipulated by users, so it shouldn't matter much in practice.